src/undo.js

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env node
+const readline = require('readline');
+const tty = require('tty');
+const chalk = require('chalk');
+
+const {
+  getConfig,
+  logUndoConfirmation,
+  logUndoConclusion
+} = require('./helpers');
+const {
+  readDirPromise,
+  rimrafPromise,
+  statPromise
+} = require('./utils');
+
+const config = getConfig();
+
+// Delete component directory
+readDirPromise(config.dir).then(files => {
+  if (files.length === 0) {
+    console.log(`There\'s no files in ${config.dir}! Is this where your components
+      are located?`);
+    process.exit(0);
+  }
+
+  let lastCreatedFile = { name: '', dateModified: 0 };
+
+  // Create promises to check which component was created last
+  const filePromises = files.map(name => {
+    return statPromise(`${config.dir}/${name}`).then(fileStat => {
+      const dateModified = fileStat.mtime.getTime();
+      if (lastCreatedFile.dateModified < dateModified) {
+        lastCreatedFile = { name, dateModified };
+      }
+    });
+  });
```

Comment:
Clever solution! Never used `fs.stat` before, looks perfect for this.

One small note: what do you think about the `statPromise.then` returning the `lastCreatedFile` object? That way, after the `Promise.all` resolves, you'll have an array of `lastCreatedFile` objects, and you can do a `.sort()` on the array to find the earliest one, using the `dateModified` as the comparator.

I think the benefit is you can make these inner functions a bit more pure, where they aren't depending on the parent scope's `lastCreatedFile` variable. It's not a huge deal because this is all enclosed in a pure function anyway, but I think it'd be slightly nicer :)

Reply:
I didn't like this solution too much, I was thinking for a neater solution too, this sounds good to me.